sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueServiceClient.java

```diff
@@ -263,7 +315,15 @@ public PagedIterable<QueueItem> listQueues() {
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
     public PagedIterable<QueueItem> listQueues(QueuesSegmentOptions options, Duration timeout, Context context) {
-        return listQueues(null, options, timeout, context);
+        Supplier<PagedIterable<QueueItem>> operation = () -> listQueues(null, options, context);
+        try {
+            return timeout != null
+                ? THREAD_POOL.submit(operation::get).get(timeout.toMillis(), TimeUnit.MILLISECONDS) : operation.get();
```

Comment:
I'm not 100% sure this is designed correctly. I believe this changed the timeout to be the creation of the `PagedIterable`, which should effectively be instant. The timeout handling needs to be moved to the get page operation, below in `listQueues(String, QueuesSegmentOptions, Context)`. 

```java
BiFunction<String, Integer, PagedResponse<QueueItem>> retriever = (nextMarker, pageSize) -> {
    if (timeout == null) {
        return this.azureQueueStorage.getServices().listQueuesSegmentSinglePage(prefix, nextMarker, 
            pageSize == null ? maxResultsPerPage : pageSize, include, null, null, finalContext);
    } else {
        return THREAD_POOL.submit(() -> this.azureQueueStorage.getServices().listQueuesSegmentSinglePage(prefix, nextMarker, 
            pageSize == null ? maxResultsPerPage : pageSize, include, null, null, finalContext)).get(timeout.toMills(), TimeUnit.MILLISECONDS);
    }
}
```